File: src/main/java/io/github/colintimbarndt/chat_emotes/data/unicode/joiner/ConstantTransformer.java
package io.github.colintimbarndt.chat_emotes.data.unicode.joiner;

import io.github.colintimbarndt.chat_emotes.data.unicode.pattern.MatchResult;
import org.jetbrains.annotations.NotNull;

public record ConstantTransformer(String value) implements MatchTransformer {
    @Override
    public void transform(@NotNull MatchResult r, @NotNull StringBuilder result) {
        result.append(value);
    }

    @Override
    public int acceptedWidth() {
        return 1;
    }
}

File: fabric_mod_server/src/main/kotlin/io/github/colintimbarndt/chat_emotes/data/unicode/joiner/ConstantTransformer.kt
package io.github.colintimbarndt.chat_emotes.data.unicode.joiner

import io.github.colintimbarndt.chat_emotes.data.unicode.pattern.MatchResult
import io.github.colintimbarndt.chat_emotes.util.StringBuilderExt.plusAssign

data class ConstantTransformer(val value: String) : MatchTransformer {
    override fun transform(r: MatchResult, result: StringBuilder) {
        result += value
    }

    override val acceptedWidth = 1
}